From: Dong Mao Zhang <dmzhang@suse.com>
Subject: Support dm-raid on large-sector devices
Patch-mainline: not yet
References: bnc#884407

On devices with a 4K sectors size, ra_disk_sb() will fail as
rdev->sb_size is less than 4K.
The buffer being read into is 1 page, so 4K is safe providing
the metadata isn't at the very end of the device, which it isn'.t

Acked-by: NeilBrown <neilb@suse.de>
Signed-off-by: Neil Brown <neilb@suse.de>

---
 drivers/md/dm-raid.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux-3.12-SLE12.orig/drivers/md/dm-raid.c
+++ linux-3.12-SLE12/drivers/md/dm-raid.c
@@ -860,7 +860,7 @@ static int super_load(struct md_rdev *rd
 	rdev->sb_start = 0;
 	rdev->sb_size = sizeof(*sb);
 
-	ret = read_disk_sb(rdev, rdev->sb_size);
+	ret = read_disk_sb(rdev, 4096);
 	if (ret)
 		return ret;
 
